homeassistant/components/amcrest/camera.py

```diff
@@ -365,10 +370,18 @@ def update(self):
                     self._brand = "unknown"
             if self._model is None:
                 resp = self._api.device_type.strip()
+                _LOGGER.debug("Device_type=%s", resp)
                 if resp.startswith("type="):
                     self._model = resp.split("=")[-1]
                 else:
                     self._model = "unknown"
+            if self._serial is None:
+                resp = self._api.serial_number.strip()
+                _LOGGER.debug("serial=%s", resp)
+                if resp.startswith("type="):
+                    self._serial = resp.split("=")[-1]
+                else:
+                    self._serial = "unknown"
```

Comment:
Since you actually does not use `self._serial` for anything other than the unique_id, this would be better:
```suggestion
            if self._attr_unique_id is None:
                resp = self._api.serial_number.strip()
                _LOGGER.debug("serial=%s", resp)
                if resp.startswith("type="):
                    self._attr_unique_id = resp.split("=")[-1]
```
Then `self._serial` and the `unique_id` property would no longer be needed

Reply:
`serial_number` is a property right, so nothing is requested here?

Reply:
It does requests in the property...

```python
    @property
    def serial_number(self) -> str:
        ret = self.command("magicBox.cgi?action=getSerialNo")
        return pretty(ret.content.decode())
```

nevermind then :smile:

Reply:
Yes, beat me to it. https://github.com/tchellomello/python-amcrest/blob/master/src/amcrest/system.py#L85